routing/missioncontrol.go

```diff
@@ -316,3 +358,58 @@ func (m *MissionControl) reportEdgeFailure(failedEdge edge,
 		minPenalizeAmt: minPenalizeAmt,
 	}
 }
+
+// GetHistorySnapshot takes a snapshot from the current mission control state
+// and actual probability estimates.
+func (m *MissionControl) GetHistorySnapshot() *MissionControlSnapshot {
+	m.Lock()
+	defer m.Unlock()
+
+	log.Debugf("Requesting history snapshot from mission control: "+
+		"node_count=%v", len(m.history))
+
+	nodes := make([]MissionControlNodeSnapshot, len(m.history))
+
+	nodeIdx := 0
+	for v, h := range m.history {
+		channelSnapshot := make([]MissionControlChannelSnapshot,
+			len(h.channelLastFail))
+
+		chanIdx := 0
+		for id, lastFail := range h.channelLastFail {
+			// Show probability assuming amount meets min
+			// penalization amount.
+			prob := m.getEdgeProbabilityForNode(
+				h, id, lastFail.minPenalizeAmt,
+			)
+
+			channelSnapshot[chanIdx] = MissionControlChannelSnapshot{
+				ChannelID:      id,
+				LastFail:       lastFail.lastFail,
+				MinPenalizeAmt: lastFail.minPenalizeAmt,
+				SuccessProb:    prob,
+			}
+
+			chanIdx++
+		}
+
+		otherProb := m.getEdgeProbabilityForNode(h, 0, 0)
+
+		nodes[nodeIdx] = MissionControlNodeSnapshot{
+			Node:                 v,
+			LastFail:             h.lastFail,
+			OtherChanSuccessProb: otherProb,
+			Channels:             channelSnapshot,
+		}
+
+		nodeIdx++
+	}
+
+	snapshot := MissionControlSnapshot{
+		Nodes: nodes,
+	}
+
+	log.Debugf("History snapshot created")
```

Comment:
Not a very useful log? Given that nothing can fail  before returning.

Reply:
Indeed, removed.